src/controllers/dlgprefcontrollers.cpp

```diff
@@ -102,14 +106,18 @@ void DlgPrefControllers::destroyControllerWidgets() {
     }
 }
 
+// TODO(Tomasito) This function is called twice at startup, find out why.
+// ...            Shouldn't it be just called once?
 void DlgPrefControllers::setupControllerWidgets() {
     // For each controller, create a dialog and put a little link to it in the
     // treepane on the left.
     QList<Controller*> controllerList =
             m_pControllerManager->getControllerList(false, true);
     qSort(controllerList.begin(), controllerList.end(), controllerCompare);
 
-    foreach (Controller* pController, controllerList) {
+    Controller* pKbdController = m_pControllerManager->getKeyboardController().data();
```

Comment:
The shared pointer is completely useless if you drop the managed reference immediately. The temporary result parameter on the stack will be destroyed implicitly before reaching the next line of code.